### PR TITLE
feat: expedition resupply notification

### PR DESCRIPTION
**Pull request by dislido**
image

### CODE REVIEW

**Comment by KagamiChan** [comment]
I think we already have many significant notice for not supplied fleets, including in-game signs:
image

and also notice in poi:
image

File: views/services/sortie-expedition-resupply-check.es [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  const fleets = getStore('info.fleets')
+  const ships = getStore('info.ships')
+  const needResupply = fleets
+    .filter((_, index) => index !== 0)
+    .map(fleet => fleet.api_ship)
+    .reduce((p, c) => {
+      p.push(...c)
+      return p
+    }, [])
```
Comment (KagamiChan) [inline views/services/sortie-expedition-resupply-check.es]: use `flatMap`

File: views/services/sortie-expedition-resupply-check.es [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    .some(shipId => {
+      const ship = ships[shipId]
+      const $ship = $ships[ship.api_ship_id]
```
Comment (KagamiChan) [inline views/services/sortie-expedition-resupply-check.es]: $ships should also come from store

**Comment by dislido** [comment]
Sometimes we don't pay so much attention to games and click the button habitually (like #1625)
so I think a toast that can block wrong clicks is necessory

File: views/components/settings/gaming/pre-sortie-config/index.es [hunk 3]
```diff
@@ -17,5 +17,12 @@ export const PreSortieConfig = withNamespaces(['setting'])(({ t }) => (
         defaultValue={true}
       />
     </FormGroup>
+    <FormGroup>
+      <SwitchConfig
+        label={t('setting:Display Expedition Resupply Notification')}
+        configName="poi.expeditionResupplyCheck.enable"
+        defaultValue={true}
```
Comment (KagamiChan) [inline views/components/settings/gaming/pre-sortie-config/index.es]: seems default value is false according to your code below

**Comment by KagamiChan** [comment]
OK, that sounds valid.
On the default value, I suggest it disabled by default

and you also need to add it to the default value config here: https://github.com/poooi/poi/blob/master/lib/default-config.es

**Comment by KagamiChan** [comment]
Thanks for your contribution.

If you're interested, feel free to take [any issue that is marked "good first issue"](https://github.com/poooi/poi/labels/good%20first%20issue), and I can give more if you'd like to help

**Pull request merged**